src/PrestaShopBundle/Form/Admin/Catalog/Category/AbstractCategoryType.php

```diff
@@ -67,51 +71,87 @@ abstract class AbstractCategoryType extends TranslatorAwareType
      */
     private $configuration;
 
+    /**
+     * @var Router
+     */
+    private $router;
+
+    /**
+     * @var CommandBusInterface
+     */
+    private $queryBus;
+
     /**
      * @param TranslatorInterface $translator
      * @param array $locales
      * @param array $customerGroupChoices
      * @param FeatureInterface $multistoreFeature
      * @param ConfigurationInterface $configuration
+     * @param Router $router
+     * @param CommandBusInterface $queryBus
      */
     public function __construct(
         TranslatorInterface $translator,
         array $locales,
         array $customerGroupChoices,
         FeatureInterface $multistoreFeature,
-        ConfigurationInterface $configuration
+        ConfigurationInterface $configuration,
+        Router $router,
+        CommandBusInterface $queryBus
     ) {
         parent::__construct($translator, $locales);
 
         $this->customerGroupChoices = $customerGroupChoices;
         $this->multistoreFeature = $multistoreFeature;
         $this->configuration = $configuration;
+        $this->router = $router;
+        $this->queryBus = $queryBus;
     }
 
     /**
      * {@inheritdoc}
      */
     public function buildForm(FormBuilderInterface $builder, array $options)
     {
+        $disableMenuThumbnailsUpload = false;
+        if (isset($options['id_category'])) {
+            $disableMenuThumbnailsUpload = $options['disable_menu_thumbnails_upload'];
+        }
+        $genericCharactersHint = $this->trans('Invalid characters:', 'Admin.Global') . ' <>;=#{}';
```

Comment:
The `<>;=#{}` should be stored in a const somewhere so it's reusable and accessible, and then you can use sprintf to generate this message

Reply:
```suggestion
        $genericCharactersHint = sprintf('%s %s', $this->trans('Invalid characters:', 'Admin.Global'), . TypedRegexValidator::CATALOG_CHARS);
```

Reply:
I'm not sure if `TypedRegexValidator::CATALOG_CHARS` is really adapted in this use case, but maybe one of its other constraints is. Regardless we should probably use this constraint here instead of a custom Regexp

Reply:
From the validator code it seems to match the regexp anyway:
https://github.com/PrestaShop/PrestaShop/blob/37f7fd4a9c5478c172263f4afec106f8686c199e/src/Core/ConstraintValidator/TypedRegexValidator.php#L108-L109

Reply:
So if you use this constraint you won't need to build the message manually